### PR TITLE
Update installation instructions to include bazel and remove outdated…

### LINKED ISSUE #4146: Update installation docs to describe how to build Ray with Bazel.

### PR DESCRIPTION
**Pull request by robertnishihara**
This fixes https://github.com/ray-project/ray/issues/4146.

### CODE REVIEW

File: doc/source/installation.rst [hunk 1]
```diff
@@ -63,23 +64,26 @@ For Ubuntu, run the following commands:
 .. code-block:: bash
 
   sudo apt-get update
-  sudo apt-get install -y cmake pkg-config build-essential autoconf curl libtool unzip flex bison psmisc python # we install python here because python2 is required to build the webui
+  sudo apt-get install -y build-essential curl unzip psmisc python # we install python here because python2 is required to build the webui
+
+  # Install Bazel.
+  ./ci/travis/install-bazel.sh
```
Comment (pcmoritz) [inline doc/source/installation.rst]: This only works once ray is cloned and once we cd'ed into the ray directory.

Comment (robertnishihara) [reply doc/source/installation.rst]: Whoops, should be fixed now. Thanks!

File: doc/source/installation.rst [hunk 2]
```diff
@@ -63,23 +65,20 @@ For Ubuntu, run the following commands:
 .. code-block:: bash
 
   sudo apt-get update
-  sudo apt-get install -y cmake pkg-config build-essential autoconf curl libtool unzip flex bison psmisc python # we install python here because python2 is required to build the webui
+  sudo apt-get install -y build-essential curl unzip psmisc python # we install python here because python2 is required to build the webui
```
Comment (pcmoritz) [inline doc/source/installation.rst]: ubuntu 14.04 also needs "git" here, but it's probably fine to not include that

File: doc/source/installation.rst [hunk 3]
```diff
@@ -63,23 +65,20 @@ For Ubuntu, run the following commands:
...
 
   # If you are not using Anaconda, you need the following.
   sudo apt-get install python-dev  # For Python 2.
   sudo apt-get install python3-dev  # For Python 3.
 
-  # If you are on Ubuntu 14.04, you need the following.
-  pip install cmake
-
   pip install cython==0.29.0
```
Comment (pcmoritz) [inline doc/source/installation.rst]: ubuntu 14.04 needs "sudo" here for the system python, but it's probably ok to assume people can figure that out (the error message is good enough)

**Review by pcmoritz — APPROVED** [review]
Tested with: Ubuntu 14.04 with Anaconda python 3.7, 16.04 with anaconda python 2.7, 18.04 with anaconda python 3.7

**Pull request merged**